deepinterpolation/generator_collection.py

```diff
@@ -897,112 +918,143 @@ class OphysGenerator(SequentialGenerator):
+            batch_frames = tf.convert_to_tensor(
+                self.movie_data[end_ind-self.batch_size:end_ind], dtype="float")
+            if not self.normalize_cache:
+                batch_frames = _normalize(batch_frames, self.local_mean, self.local_std)
+            self._data_tensor = tf.concat(
+                [self._data_tensor[self.batch_size:], batch_frames], 0)
+            self.cached_index = index
+        
+        elif self.cached_index == index:
+            return self._data_tesnor
         
-        if self.cache_data:
-            self.raw_data = ( self.raw_data.astype("float") \
-                - self.local_mean ) / self.local_std
-
-        movie_obj_point.close()
-
-    def __getitem__(self, index):
-        shuffle_indexes = self.generate_batch_indexes(index)
-
-        input_full = np.zeros(
-            [self.batch_size, 512, 512, self.pre_frame + self.post_frame],
-            dtype="float32",
-        )
-
-        output_full = np.zeros([self.batch_size, 512, 512, 1], dtype="float32")
-
-        for batch_index, frame_index in enumerate(shuffle_indexes):
-            X, Y = self.__data_generation__(frame_index)
-            X_shape = X.shape
-            Y_shape = Y.shape
-
-            input_full[batch_index, : X_shape[0], : X_shape[1], :] = X
-            output_full[batch_index, : Y_shape[0], : Y_shape[1], 0] = Y
-
-        return input_full, output_full
-
-    def __data_generation__(self, index_frame):
-        "Generates data containing batch_size samples"
-
-        if self.cache_data:
-            movie_obj = self.raw_data
         else:
-            movie_obj_point = h5py.File(self.raw_data_file, "r")
-            movie_obj = movie_obj_point['data']
+            self._data_tensor = tf.convert_to_tensor(
+                self.movie_data[start_ind:end_ind], dtype="float")
+            self.cached_index = index
+            if not self.normalize_cache:
+                self._data_tensor = _normalize(self._data_tensor, self.local_mean, self.local_std)
+        return self._data_tesnor
 
-        input_index = np.arange(
-            index_frame - self.pre_frame - self.pre_post_omission,
-            index_frame + self.post_frame + self.pre_post_omission + 1,
-        )
-        input_index = input_index[input_index != index_frame]
-
-        for index_padding in np.arange(self.pre_post_omission + 1):
-            input_index = input_index[input_index !=
-                                      index_frame - index_padding]
-            input_index = input_index[input_index !=
-                                      index_frame + index_padding]
 
-        # If data was cached we do not need to normalize. this was done 
-        # at once to minimize compute
-        if self.cache_data:
-            data_img_input = movie_obj[input_index, :, :]
-            data_img_output = movie_obj[index_frame, :, :]
+    def __getitem__(self, index):
+        if not (self._gpu_available and not self.gpu_cache_full):
```

Comment:
This is checking if not self._gpu_available and self.gpu_cache_full ? That should never be true?